pennylane/qnode.py

```diff
@@ -592,6 +592,16 @@ def construct(self, args, kwargs):
 
     def __call__(self, *args, **kwargs):
 
+        for pos_arg in args:
+            if not isinstance(pos_arg, qml.numpy.tensor):

```

Comment:
This warning should only be raised when using the autograd interface, _not_ pytorch/jax/tf!

Reply:
In addition, I believe it should check:

1. Whether the positional argument requires grad (via `qml.math.requires_grad`)
2. Only then, if it is not a `np.tensor`, should the warning be raised.